### PR TITLE
🎉 Python 3.10 support

### PR DESCRIPTION
**Pull request by misl6**
Switches to **Python 3.10.10** for both **python3** and **hostpython3**

- Removed `_sqlite3` mock module setup. This is not needed anymore 🥳 🎉 
- Added `mmap` support (Closes https://github.com/kivy/kivy-ios/issues/659)
- Removed `host_setuptools`(from the Python2 era) and `host_setuptools3`. This is not needed anymore as `hostpython3` now provides `setuptools` (which we are patching, due to a possible bug in `distutils` codebase)
- Tested during runtime. Looks faster 🏎️ 
- ⚠️ Needs some testing, and we will likely need to upgrade some recipes, for that reason we will need to stay away from releases for a while⚠️

### CODE REVIEW

File: kivy_ios/recipes/hostpython3/allow-cflags-override.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+--- setuptools.orig/_distutils/sysconfig.py	2023-02-18 18:21:53
++++ setuptools/_distutils/sysconfig.py	2023-02-18 18:23:17
+@@ -329,7 +329,9 @@
+         if 'LDFLAGS' in os.environ:
+             ldshared = ldshared + ' ' + os.environ['LDFLAGS']
+         if 'CFLAGS' in os.environ:
+-            cflags = cflags + ' ' + os.environ['CFLAGS']
++            # cflags = cflags + ' ' + os.environ['CFLAGS']
++            # TODO: we may have found a bug in distutils?
```
Comment (AndreMiras) [inline kivy_ios/recipes/hostpython3/allow-cflags-override.patch:9]: Is that a TODO for an upstream bug report?

**Review by AndreMiras — APPROVED** [review]
Awesome! :100:

File: kivy_ios/recipes/hostpython3/allow-cflags-override.patch (hunk 1 above)
Comment (misl6) [reply kivy_ios/recipes/hostpython3/allow-cflags-override.patch:9]: Yeah, this is a personal reminder to check blame history and open a bug report on `distutils` upstream if needed 😅

Comment (misl6) [reply kivy_ios/recipes/hostpython3/allow-cflags-override.patch:9]: (Hopefully will do next weekend)

**Pull request merged**